### PR TITLE
Issue67

### PR DESCRIPTION
**Pull request by derevnja**

### CODE REVIEW

File: easy_profiler_converter/converter.cpp [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    if(!m_file_out.empty())
+    {
+        std::ofstream file(m_file_out);
+        file << json;
+    }
+    else
+    {
+        ::std::cout << nlohmann::json(json).dump(2);
+    }
+    return;
```
[the remark this answers is not in the document]
Comment (derevnja) [reply easy_profiler_converter/converter.cpp]: why do u think this stuff is useless?

[the remark this answers is not in the document]
Comment (yse) [reply easy_profiler_converter/converter.cpp]: return from void at the end of function =)

File: easy_profiler_core/reader.cpp [hunk 2]
```diff
@@ -523,6 +524,7 @@ extern "C" {
         serialized_descriptors.set(descriptors_memory_size);
         //validate_pointers(progress, olddata, serialized_descriptors, descriptors, descriptors.size());
 
+        std::cout << "!!!" << total_descriptors_number << "!!!";
```
[the remark this answers is not in the document]
Comment (derevnja) [reply easy_profiler_core/reader.cpp]: ok

File: easy_profiler_converter/converter.cpp (hunk 1 above)
[the remark this answers is not in the document]
Comment (derevnja) [reply easy_profiler_converter/converter.cpp]: ok

File: easy_profiler_converter/converter.h [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+#include "reader.h"
+
+///nlohmann json
+#include "include/json.hpp"
+
+class JSONConverter EASY_FINAL
+{
+public:
+    JSONConverter(const ::std::string &file_in,
+                  const ::std::string &file_out):
```
Comment (yse) [inline easy_profiler_converter/converter.h]: Not so clear interface. If `file_out` is empty string output will be dumped in `stdout`. May be we should not use classes. Think about more clear interface

File: easy_profiler_core/include/easy/easy_protocol.h [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+
+///for actual version vistit https://github.com/yse/easy_profiler/wiki/.prof-file-format-v1.3.0
+
+namespace profiler {
+
+namespace reader {
+
+struct BlockDescriptor;
+
+struct BlocksInfo         //12
```
Comment (yse) [inline easy_profiler_core/include/easy/easy_protocol.h]: What is 12? =)

File: easy_profiler_converter/reader.cpp [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+typedef ::std::unordered_map<::profiler::hashed_stdstring, ::profiler::block_id_t> IdMap;
+typedef ::std::unordered_map<::profiler::hashed_stdstring, ::profiler::BlockStatistics*> CsStatsMap;
+
+#endif
+
+/// end from easy_profiler_core/reader.cpp/////
+
+using namespace profiler::reader;
+
+::profiler::block_index_t FileReader::readFile(const ::std::string &filename)
```
Comment (yse) [inline easy_profiler_converter/reader.cpp]: Let's split this function on several functions

Comment (derevnja) [reply easy_profiler_converter/reader.cpp]: ok. good idea, shall try.

File: easy_profiler_converter/converter.h (hunk 3 above)
Comment (derevnja) [reply easy_profiler_converter/converter.h]: accepted.

**Comment by rationalcoder** [comment]
Wait, what is the small project trying to accomplish. Is it trying to perform what I was talking about in #67 ?
If so, the API for this kind of thing would need to be a lighter-weight c-api. That way, it would be trivial to load exporters as plugins and provide the "Export with" menu option.

**Comment by derevnja** [comment]
Hi!
At the moment I really don't know for sure about of the real need for class-way API.
So, I've already been told to reimliment the API in a kind of c-api.
But for me this is not so clear yet.
Anyway, I think if we start with command line converter we can add it as a plugin anytime.

**Comment by derevnja** [comment]
Hi!
At the moment I really don't know for sure about of the real need for class-way API.
So, I've already been told to reimliment the API in a kind of c-api.
But for me this is not so clear yet.
Anyway, I think if we start with command line converter we can add it as a plugin anytime.

**Comment by derevnja** [comment]
Hi!
At the moment I really don't know for sure about of the real need for class-way API.
So, I've already been told to reimliment the API in a kind of c-api.
But for me this is not so clear yet.
Anyway, I think if we start with command line converter we can add it as a plugin anytime.

**Pull request merged**